### PR TITLE
DSPHLE: Seperate HLE construction and initialization.

### PR DESCRIPTION
**Pull request by degasus**
We must be able to construct the HLE without emulating the initialization
on loading a savestate.

### CODE REVIEW

**Comment by JMC47** [comment]
This fixes Dolphin or the game crashing when loading savestates shortly after booting.

**Comment by Tilka** [comment]
lgtm

File: Source/Core/Core/HW/DSPHLE/UCodes/AX.cpp [hunk 1]
```diff
@@ -18,17 +18,21 @@
...
 }
 
 AXUCode::~AXUCode()
 {
   m_mail_handler.Clear();
 }
 
+void AXUCode::Initialize()
+{
+  m_mail_handler.PushMail(DSP_INIT);
```
Comment (delroth) [inline Source/Core/Core/HW/DSPHLE/UCodes/AX.cpp]: Can you maybe try replacing these two lines with PushMail(DSP_INIT, true)?

This has smarter interrupt scheduling behavior in the case of messages being queued.

Comment (Tilka) [inline Source/Core/Core/HW/DSPHLE/UCodes/AX.cpp]: Let's do that in the next PR.

Comment (degasus) [inline Source/Core/Core/HW/DSPHLE/UCodes/AX.cpp]: Yeah, let's try to not mix such a behavior change.
Isn't this change exactly what you did in PR #3756 ? Do you want me to rebase and to open a new PR for it?

**Pull request merged**